src/checkout-buttons/strategies/apple-pay/apple-pay-button-strategy.ts

```diff
@@ -0,0 +1,397 @@
+import { RemoteCheckoutActionCreator } from '../../../remote-checkout';
+import { ConsignmentActionCreator, ShippingOption } from '../../../shipping';
+import { CheckoutButtonInitializeOptions } from '../../checkout-button-options';
+import CheckoutButtonStrategy from '../checkout-button-strategy';
+
+const validationEndpoint = (bigPayEndpoint: string) => `${bigPayEndpoint}/api/public/v1/payments/applepay/validate_merchant`;
+
+enum DefaultLabels {
+    Subtotal = 'Subtotal',
+    Shipping = 'Shipping',
+}
+
+function isShippingOptions(options: ShippingOption[] | undefined): options is ShippingOption[] {
+    return options instanceof Array;
+}
+
+export default class ApplePayButtonStrategy implements CheckoutButtonStrategy {
+    private _paymentMethod?: PaymentMethod;
+    private _applePayButton?: HTMLElement;
+    private _onAuthorizeCallback = noop;
+    private _onError = noop;
+    private _subTotalLabel: string = DefaultLabels.Subtotal;
+    private _shippingLabel: string = DefaultLabels.Shipping;
+
+    constructor(
+        private _store: CheckoutStore,
+        private _checkoutActionCreator: CheckoutActionCreator,
+        private _requestSender: RequestSender,
+        private _paymentMethodActionCreator: PaymentMethodActionCreator,
+        private _consignmentActionCreator: ConsignmentActionCreator,
+        private _billingAddressActionCreator: BillingAddressActionCreator,
+        private _paymentActionCreator: PaymentActionCreator,
+        private _remoteCheckoutActionCreator: RemoteCheckoutActionCreator,
+        private _orderActionCreator: OrderActionCreator,
+        private _sessionFactory: ApplePaySessionFactory
+    ) {}
+
+    async initialize(options: CheckoutButtonInitializeOptions): Promise<InternalCheckoutSelectors> {
+
+        const { methodId, containerId , applepay}  = options;
+
+        assertApplePayWindow(window);
+
+        if (!methodId || !applepay) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        const {
+            buttonClassName,
+            shippingLabel,
+            subtotalLabel,
+            onError = () => {},
+            onPaymentAuthorize,
+        } = applepay;
+
+        this._shippingLabel = shippingLabel || DefaultLabels.Shipping;
+        this._subTotalLabel = subtotalLabel || DefaultLabels.Subtotal;
+        this._onAuthorizeCallback = onPaymentAuthorize;
+        this._onError = onError;
+
+        const state = await this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(methodId));
+        this._paymentMethod = state.paymentMethods.getPaymentMethodOrThrow(methodId);
+
+        this._applePayButton = this._createButton(containerId, buttonClassName);
+        this._applePayButton.addEventListener('click', this._handleWalletButtonClick);
+
+        return this._store.getState();
+    }
+
+    deinitialize(): Promise<InternalCheckoutSelectors> {
```

Comment:
Should be able to return `Promise.resolve();` here and that way we don't need to change the base interface.

Reply:
Yes, I have changed the base interface back and tested the code, works good.